### PR TITLE
Minor coremark fixes and build under meson

### PR DESCRIPTION
**Pull request by GregAC**
This allows you to build coremark with meson. I have tested the resulting the binaries on both FPGA and verilator.

Still some things to do (need different compiler flags to get best performance that currently cause coremark to crash on FPGA for unknown reasons and the functionality that gives us time in seconds doesn't seem to be giving correct results on FPGA) but this is a functional starting point.

### CODE REVIEW

File: sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c [hunk 1]
```diff
@@ -480,7 +480,12 @@ static int ee_vsprintf(char *buf, const char *fmt, va_list args) {
 
   for (str = buf; *fmt; fmt++) {
     if (*fmt != '%') {
-      *str++ = *fmt;
+      if(*fmt == '\n') {
```
Comment (imphil) [inline sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c]: nit: `if (`

Please run this file through clang-format (all files except for the vendor files)

**Comment by imphil** [comment]
And you need Signed-Off-By lines on all commits.

File: sw/device/benchmarks/coremark/README.md [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
-Controls how many iteartions are run.  By default this value is 0 and will cause coremark to run >10s.
-For simulation and verilator, set to 1 for a reduced run
+To build coremark under meson:
 
-## For verilator
-make distclean; make SIM=1 ITERATIONS=1
-SIM=1 matces the verilator UART baudrates
+```sh
+cd "${REPO_TOP}"
+./meson_init.sh -f
```
Comment (imphil) [inline sw/device/benchmarks/coremark/README.md]: I guess in anticipation of https://github.com/lowRISC/opentitan/pull/1040 you can already skip the `-f` here.

File: sw/device/benchmarks/coremark/meson.build [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+              coremark_base_files],
+    name_suffix: 'elf',
+    dependencies: [
+      sw_lib_uart,
+      sw_lib_mem,
+      riscv_crt,
+    ],
+    c_args : ['-DPERFORMANCE_RUN=1', iterations, '-DTOTAL_DATA_SIZE=2000', '-DMAIN_HAS_NOARGC=1']
+  ),
+)
```
Comment (imphil) [inline sw/device/benchmarks/coremark/meson.build]: Can you follow the template meson file e.g. in hello_world to have an export target available?

File: sw/device/benchmarks/coremark/README.md [hunk 4]
```diff
@@ -1,13 +1,28 @@
...
+ninja -C build-out/sw/${TARGET} sw/device/benchmarks/coremark/coremark.bin
```
Comment (imphil) [inline sw/device/benchmarks/coremark/README.md]: This will need adjustment if you use the common meson file template, it could then read

```
ninja -C build-out/sw/${TARGET} sw/device/benchmarks/coremark/coremark_export
```

File: sw/device/benchmarks/coremark/README.md [hunk 5]
```diff
@@ -1,13 +1,28 @@
...
+```
 
-## For DV / FPGA
-make distclean; make ITERATIONS=1/N
+Where ${TARGET} is one of 'sim-verilator' or 'fpga'
+
+This will give you a .bin and .vmem file (suitable for either spiflash or
+giving directly to --flashinit for verilator) which can be found in
+`build-out/sw/${TARGET}/sw/device/benchmarks/coremark`
```
Comment (imphil) [inline sw/device/benchmarks/coremark/README.md]: Please don't rely on any binary path in build-out. Use the ones in build-bin instead.

File: sw/device/benchmarks/coremark/README.md [hunk 6]
```diff
@@ -1,13 +1,28 @@
-# Options
-There are two main options available for coremark SIM, ITERATIONS
+# Building Coremark
```
Comment (imphil) [inline sw/device/benchmarks/coremark/README.md]: Coremark -> CoreMark

File: sw/device/benchmarks/coremark/README.md [hunk 7]
```diff
@@ -1,13 +1,28 @@
...
 
-## ITERATIONS
-Controls how many iteartions are run.  By default this value is 0 and will cause coremark to run >10s.
-For simulation and verilator, set to 1 for a reduced run
+To build coremark under meson:
```
Comment (imphil) [inline sw/device/benchmarks/coremark/README.md]: coremark -> CoreMark
meson -> Meson

File: sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c [hunk 8]
```diff
@@ -663,6 +663,10 @@ int ee_printf(const char *fmt, ...) {
   va_end(args);
   p = buf;
   while (*p) {
+    if (*p == '\n') {
+      uart_send_char('\r');
+    }
```
Comment (imphil) [inline sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c]: Why is that needed?

Comment (GregAC) [reply sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c]: The plain \n going over the serial results in the next line being immediately below the end of the previous one. \r is required to return to the beginning of the line hence a '\n' -> '\r\n' transformation is needed.

I did initially do this in ee_vsprintf but felt it made more sense here (as it's only an artifact is sending the string to the uart so the transformation should be done there rather than on the string generated by ee_vsprintf).

Comment (imphil) [reply sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c]: That behavior probably depends on your serial terminal viewer, but good enough.

Comment (GregAC) [reply sw/device/benchmarks/coremark/top_earlgrey/ee_printf.c]: True, screen definitely wants it. I think it doesn't matter to include it when it's not needed
